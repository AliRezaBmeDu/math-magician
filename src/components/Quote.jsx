import React, { useState, useEffect } from 'react';
import spinLogo from './arrow-clockwise.svg';

const Quote = () => {
  const [data, setData] = useState([]);
  const [isLoading, setIsLoading] = useState(true);
  const apiKey = 'vA0/JB/b2F3uIe+n+OUoAA==ssQu65YPz7YnNSOQ';

  const fetchQuoteData = async () => {
    try {
      const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=happiness', {
        headers: {
          'X-Api-Key': apiKey,
        },
      });
      const json = await response.json();
      setData(json);
      console.log('JSON response: ', json);
      console.log('data: ', data);
    } catch (error) {
      console.error('Request failed:', error);
    }
    setIsLoading(false);
  };

  useEffect(() => {
    fetchQuoteData();
  }, []);

  if (isLoading) {
    return (
      <div className="quote-container">
        <div>Loading...</div>
      </div>
    );
  }

  return (
    <div className="quote-container">
      <div className="quote"><p>{ data[0].quote }</p></div>
      <div className="author">{ data[0].author }</div>
      <button type="button" className="reload-button">
        <img src={spinLogo} alt="reload-icon" />
        <p>Refresh</p>
      </button>
    </div>
  );
};

export default Quote;
