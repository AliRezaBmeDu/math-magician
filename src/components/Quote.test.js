import React from 'react';
import renderer from 'react-test-renderer';
import Quote from './Quote';

// Mock the fetch function and provide a mock implementation
global.fetch = jest.fn(() => Promise.resolve({
  json: async () => [{ quote: 'Mocked Quote', author: 'Mocked Author' }],
}));

it('should render without crashing', async () => {
  const component = renderer.create(<Quote />);
  await component.root.findByType(Quote); // Wait for async actions to complete
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
