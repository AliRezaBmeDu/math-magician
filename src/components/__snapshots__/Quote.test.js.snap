// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="quote-container"
>
  <div
    className="quote-author"
  >
    <p>
      Loading...
    </p>
  </div>
</div>
`;
